pkg/controller/networkpolicy/crd_utils.go

```diff
@@ -54,6 +54,32 @@ func toAntreaServicesForCRD(npPorts []v1alpha1.NetworkPolicyPort) ([]controlplan
 	return antreaServices, namedPortExists
 }
 
+// toAntreaServicesAndPeersFromServiceReference converts v1alpha1.ServiceReference
+// object to a slice of Antrea Service objects and a slice of Antrea controlplane
+// NetworkPolicyPeer. Antrea Service here means backend Endpoints Ports.
+func (n *NetworkPolicyController) toAntreaServicesAndPeersFromServiceReference(serviceReference *v1alpha1.ServiceReference) ([]controlplane.Service, *controlplane.NetworkPolicyPeer, error) {
+	var antreaServices []controlplane.Service
+	var antreaPeers controlplane.NetworkPolicyPeer
+
+	endpoints, err := n.endpointsLister.Endpoints(serviceReference.Namespace).Get(serviceReference.Name)
```

Comment:
I have a question: For NodePort Service, should we install rules matching the Endpoints or just skip it?

Reply:
My preferred solution would be to install the rules for all Services, but add a log message if the Service is NodePort. Of course you don't want to log the same message too many times so that's one thing to look out for.

Reply:
Are you worried that every time we reprocess the ACNP with NodePort Service in toServices, there will be a log?